scripts/mk.js

```diff
@@ -16034,6 +16059,28 @@ function move(getId, triggered) {
 				oKart.speed = oKart.maxspeed*cappedRelSpeed(oKart);
 				playIfShould(oKart,"musics/events/boost.mp3");
 			}
+			else if ((touche_etoile(fNewPosX, fNewPosY) || (fSelectedClass > 1.5 && touche_etoile(fMidPosX, fMidPosY))) && !oKart.tourne) {
+				for (var i=0;i<strPlayer.length;i++)
+					oKart.sprite[i].img.src = getStarSrc(oKart.personnage);
+				if (!oKart.cpu && !oKart.etoile) {
+					if (!isOnline) {
+						oKart.sprite[0].img.onload = function() {
+							bCounting = false;
+							this.onload = undefined;
+							reprendre(false);
+						}
+						interruptGame();
+						bCounting = true;
+					}
+					if (shouldPlaySound(oKart) && !oPlayers[1])
+						postStartMusic("musics/events/starman.mp3");
+				}
+				if (oKart.speedinc > 0)
+					oKart.speedinc *= 5;
+				delete oKart.shift;
+				oKart.protect = true;
+				oKart.etoile = 80;
```

Comment:
Can we refactor this code part into a common function and call it in the 2 places? I prefer to avoid copy & pasted code, it makes it harder to maintain afterwards

Reply:
I added a bullet bill check on the `touche_etoile` line and also added a `setStarState` function to avoid copy-pasting the star code. I made the function take the star duration as a parameter (just in case we need a shorter duration for ground-picked stars)